### ISSUE TITLE
[TearsheetNarror focus]: Opening MenuButton messes up with focus when there is also other button on the tearsheet narrow

### ISSUE THREAD

**Issue by nelsonchen90** [carbon-design-system/ibm-products#5957]
### Package

Carbon for IBM Products

### Description

- corner case of MenuButton from carbon/react and TearsheetNarrow from ibm-product
- when user clicks/open the MenuButton dropdown from a TearsheetNarrow, the focus will be locked to the first focusable element on the tearsheet, in my case a button.
- User is not able to 'unset' the focus and scrolling behavior is weird.

https://github.com/user-attachments/assets/3daba8bb-fb0b-49c1-b724-a1caeef9f6fd

### Component(s) impacted

- MenuButton (@carbon/react)
- html button
- TearsheetNarrow

### Browser

Chrome

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

v2.47.0

### Suggested Severity

Severity 1 = The design is broken in a critical way that blocks users from completing tasks or damages the brand. Affects major functionality, no workaround.

### Product/offering

IBM knowledge catalog / IBM Software

### CodeSandbox or Stackblitz example

https://stackblitz.com/edit/github-mtapvf?file=package-lock.json

### Steps to reproduce the issue (if applicable)

In the stackblitz. 
Try the same component on the tearsheet and outside of the tearsheet. 
You will see the focus is forced on the button when the MenuButton is open on the tearsheet. and user is not able to scroll any more on tearsheet.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

**Comment by elycheea** [comment]
@makafsal This might be another one related to `useFocus()` if we want to review a few of these related issues collectively in the next sprint? 🤔

**Comment by makafsal** [comment]
@elycheea Focus is working fine, the issue can't reproduce locally.

**Comment by makafsal** [comment]
@elycheea I don't think this is entirely related to `useFocus()` hook. There are some calculations happening in `MenuButton`, I'm still debugging it 😇

**Comment by makafsal** [comment]
@elycheea I'm able to reproduce this locally by setting `'enable-experimental-focus-wrap-without-sentinels': false`, so I expect this issue will resolve in the upcoming release due to this PR: https://github.com/carbon-design-system/ibm-products/pull/5882

**Comment by nelsonchen90** [comment]
@makafsal thanks! is there ETA on upcoming release that would include that? we have release coming up next week and i am trying to see if we can get this in in time or we will have to document a known issue.

**Comment by nelsonchen90** [comment]
@elycheea maybe you can help answer that question above? thanks

**Comment by elycheea** [comment]
@nelsonchen90 We just released on Wednesday so it’ll be in the _next_ release. You can see our release schedule [here](https://github.com/carbon-design-system/ibm-products/wiki/Carbon-for-IBM-Products-Releases#schedule).

Our next code freeze will be this coming Tuesday though so you should be able to at least test the fix using the release candidate. 🙏

**Comment by matthewgallo** [comment]
Will be included in release today.
